grails-app/controllers/org/transmartproject/rest/StudyController.groovy

```diff
@@ -27,25 +27,42 @@ package org.transmartproject.rest
 
 import grails.rest.Link
 import grails.rest.render.util.AbstractLinkingRenderer
+import org.springframework.beans.factory.annotation.Autowired
+import org.transmartproject.db.accesscontrol.AccessControlChecks
+import org.transmartproject.rest.misc.CurrentUser
 
 import javax.annotation.Resource
 
 import org.transmartproject.core.ontology.StudiesResource
 import org.transmartproject.core.ontology.Study
 import org.transmartproject.rest.marshallers.ContainerResponseWrapper
 
+import static org.transmartproject.core.users.ProtectedOperation.WellKnownOperations.READ
+
 class StudyController {
 
     static responseFormats = ['json', 'hal']
 
     @Resource
     StudiesResource studiesResourceService
 
+    @Autowired
+    AccessControlChecks accessControlChecks
+
+    @Autowired
+    CurrentUser currentUser
+
     /** GET request on /studies/
      *  This will return the list of studies, where each study will be rendered in its short format
     */
     def index() {
-        respond wrapStudies(studiesResourceService.studySet)
+        def studiesImpl =  wrapStudies(studiesResourceService.studySet)
```

Comment:
Did you try first decorating the studies with your access field and only after that wrapping them? This way you could avoid unwrapping the studies here.